pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -287,122 +204,16 @@ func (h *appHandler) statusAggregate(appFile *appfile.Appfile) ([]common.Applica
-	compName := comp.GetName()
-	compNameSpace := comp.GetNamespace()
-	compKey := ctypes.NamespacedName{Name: compName, Namespace: compNameSpace}
-
-	err := h.r.Get(ctx, compKey, &curComp)
-	if err != nil {
-		if !apierrors.IsNotFound(err) {
-			return "", err
-		}
-		if err = h.r.Create(ctx, comp); err != nil {
-			return "", err
-		}
-		klog.InfoS("Created a new component", "component", klog.KObj(comp))
-	} else {
-		// remember the revision if there is a previous component
-		if curComp.Status.LatestRevision != nil {
-			preRevisionName = curComp.Status.LatestRevision.Name
-		}
-		comp.ResourceVersion = curComp.ResourceVersion
-		if err := h.r.Update(ctx, comp); err != nil {
-			return "", err
-		}
-		klog.InfoS("Updated a component", "component", klog.KObj(comp))
-	}
-	// remove the object from the raw extension before we can compare with the existing componentRevision whose
-	// object is persisted as Raw data after going through api server
-	updatedComp := comp.DeepCopy()
-	updatedComp.Spec.Workload.Object = nil
-	if updatedComp.Spec.Helm != nil {
-		updatedComp.Spec.Helm.Release.Object = nil
-		updatedComp.Spec.Helm.Repository.Object = nil
-	}
-	if len(preRevisionName) != 0 {
-		needNewRevision, err := utils.CompareWithRevision(ctx, h.r, compName, compNameSpace, preRevisionName, &updatedComp.Spec)
-		if err != nil {
-			return "", errors.Wrap(err, fmt.Sprintf("compare with existing controllerRevision %s failed",
-				preRevisionName))
-		}
-		if !needNewRevision {
-			klog.InfoS("No need to wait for a new component revision", "component", klog.KObj(updatedComp),
-				"revision", preRevisionName)
-			return preRevisionName, nil
-		}
-	}
-	klog.InfoS("Wait for a new component revision", "component name", compName,
-		"previous revision", preRevisionName)
-	// get the new component revision that contains the component with retry
-	checkForRevision := func() (bool, error) {
-		if err := h.r.Get(ctx, compKey, &curComp); err != nil {
-			// retry no matter what
-			// nolint:nilerr
-			return false, nil
-		}
-		if curComp.Status.LatestRevision == nil || curComp.Status.LatestRevision.Name == preRevisionName {
-			return false, nil
-		}
-		needNewRevision, err := utils.CompareWithRevision(ctx, h.r, compName,
-			compNameSpace, curComp.Status.LatestRevision.Name, &updatedComp.Spec)
-		if err != nil {
-			// retry no matter what
-			// nolint:nilerr
-			return false, nil
-		}
-		// end the loop if we find the revision
-		if !needNewRevision {
-			curRevisionName = curComp.Status.LatestRevision.Name
-			klog.InfoS("Get a matching component revision", "component name", compName,
-				"current revision", curRevisionName)
-		}
-		return !needNewRevision, nil
-	}
-	if err := wait.ExponentialBackoff(utils.DefaultBackoff, checkForRevision); err != nil {
-		return "", err
-	}
-	return curRevisionName, nil
-}
-
-func (h *appHandler) applyHelmModuleResources(ctx context.Context, comp *v1alpha2.Component, owners []metav1.OwnerReference) error {
```

Comment:
why deleted

Reply:
rfc it into applying workload addon resources.